apps/battery/app.js

```diff
@@ -1,15 +1,352 @@
+              h = Math.floor(data.batTimetoEmpty / 60)
+              m = data.batTimetoEmpty % 60
+              text = util.format(constant.batteryDisconnect19, h, m)
+            }
+          }
+          notifyTTS(text)
+        })
+      }
+    })
+  }
+
+  function notifyMedia (url, callback) {
+    logger.log('notify media will setForeground:', url)
+    activity.setForeground().then(() => {
+      logger.log('notify media setForeground end will start media:', url)
+      activity.media.start(url).then(() => {
+        if (typeof (callback) === 'function') {
+          callback()
+          return
+        }
+        logger.log('notify media callback will setBackground:', url)
+        activity.setBackground()
+      }).catch(error => {
+        logger.warn(error)
+      })
+    })
+  }
+
+  function lowerPower (percent, isPlaying) {
+    var url
+    if (percent === 10) {
+      url = resourcePath.lowPower10
+    } else if (percent === 20) {
+      if (isPlaying && isPlaying === 'true') {
+        url = resourcePath.lowPower20Play
+      } else {
+        url = resourcePath.lowPower20Idle
+      }
+    }
+    notifyMedia(url)
+  }
+
+  function pushNotification () {
+    var date = new Date()
+    var h = date.getHours()
+    var content
+    if (h >= 22 || h <= 7) { // push消息：电量量低于10%，帮我充上电就早点休息 吧，迎接元⽓气满满的⼀一天!
+      content = constant.notificationNight
+    } else { // push消息：“电量量低于10%，我不不想⾃自动关机⽽而断了了与你的联系，快去帮 我充电或者打开侧⾯面开关进⼊入休眠模式帮我省电吧” / “电量量低于10%，世界上最遥远的距离就是你 迟迟不不来找我，⽽而我却在痴痴地等着你为我“续命”。”(两句句随机出）
+      var seed = Math.random()
+      logger.error(seed, seed > 0.5 ? constant.notification1 : constant.notification2)
+      content = seed > 0.5 ? constant.notification1 : constant.notification2
+    }
+    var body = {
+      'message': content,
+      'extra': '{"sys":{}}'
+    }
+    var bodyStr = JSON.stringify(body)
+    logger.log('pushNotification:', bodyStr)
+    activity.httpgw.request(constant.urls.PUSH_MOBILE_MSG, body, { services: 'rest' }).then((res) => {
+      logger.log('pushNotification result:', bodyStr, res)
+    })
+  }
+
+  function temperatureAbnormal (isHighTemperature) {
+    logger.warn('temperatureAbnormal:', isHighTemperature)
+    var url
+    if (isHighTemperature) {
+      url = resourcePath.temperature50
+    } else {
+      url = resourcePath.temperature0
+    }
+    notifyMedia(url)
+  }
+
+  var temperatureTimeId
+  function pollingCheckTemperature () {
+    if (temperatureTimeId) {
+      logger.warn('temperature check timer is started')
+      return
+    }
+    temperatureTimeId = setInterval(function () {
+      // check temperature if not safe will notifyLight again or safe will cancel timer
+      logger.warn('temperature timer callback will check again')
+      queryBatteryStatus().then(data => {
+        if (data.batTemp >= 55 || data.batTemp <= 0) {
+          activity.light.play(TEMPERATURE_LIGHT_RES)
+        } else {
+          clearInterval(temperatureTimeId)
+        }
+      })
```

Comment:
I think we should handle the error when queryBatteryStatus failed, which clear the interval handle, either.